### PR TITLE
dev: Sites & Apps dark mode support

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-1028

## Proposed changes

* Adds dark mode support for sites & apps
* Widget/type view, alpha view

## Reviewer notes

Remember to check both sites & apps by type, and by alphabetical order.

## Setup

* Run both the portal and cms
* Set the `theme` key to `dark` in localStorage

### As the original developer, I have

- [x] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [x] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [x] On commercial internet in Edge
  - [x] On commercial internet in Firefox
  - [x] On commercial internet in Chrome
  - [x] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

### CODE REVIEW

File: src/styles/pages/sitesAndApplications.module.scss [hunk 1]
```diff
@@ -26,25 +26,125 @@
...
+    :global(.sitesAndAppsBookmark) {
+      background-color: var(--bookmark-sites-apps-bg);
+      border-color: var(--bookmark-sites-apps-border);
+    }
+  }
 }
 
+[data-color-theme='dark'] {
+  .widgetContainer {
+    border: none;
```
Comment (jcbcapps) [inline src/styles/pages/sitesAndApplications.module.scss:60]: I think there should be a border here? In Figma there is a border with what looks like a transparent bg for the container. Apologies if this was discussed elsewhere and I missed it!

Comment (abbyoung) [reply src/styles/pages/sitesAndApplications.module.scss:60]: thanks -- i'll take a look!

**Review by jbecker01 — APPROVED** [review]
👍

File: src/styles/pages/sitesAndApplications.module.scss (hunk 1 above)
Comment (abbyoung) [reply src/styles/pages/sitesAndApplications.module.scss]: fixed 🌠

**Review by jcbcapps — APPROVED** [review]

File: src/components/BookmarkList/BookmarkList.tsx [hunk 2]
```diff
@@ -97,39 +97,43 @@ type BookmarkListProps = {
...
-            bookmark={b}
-            userCollectionOptions={userCollectionOptions}
-            handleAddToCollection={handleAddToCollection}
-            canAddNewCollection={canAddNewCollection}
-          />
-        ))}
-      </tbody>
-    </Table>
+    <div className={className}>
+      <Table striped fullWidth>
```
Comment (gidjin) [inline src/components/BookmarkList/BookmarkList.tsx:116]: question: I'm curious why did we need a new div here vs applying className to `Table`?

File: src/components/Collection/Collection.tsx [hunk 3]
```diff
@@ -9,12 +9,19 @@ type PropTypes = {
...
+  title = '',
+  children,
+  header,
+  footer,
+  className = '',
+}: PropTypes) => {
   return (
     <Widget
-      className={styles.collection}
+      className={className}
```
Comment (gidjin) [inline src/components/Collection/Collection.tsx:24]: Question: I'm not sure but do we need to combine these classes? Or does this make `src/components/Collection/Collection.module.scss` unnecessary?

**Review by gidjin — COMMENTED** [review]
question: Should the left navigation be included with this PR? Or is there another story that would be part of since it looks like it should be green in figma.

SCR-20221004-o94

**Comment by abbyoung** [comment]
> question: Should the left navigation be included with this PR? Or is there another story that would be part of since it looks like it should be green in figma.
> 
> SCR-20221004-o94

left nav is taken care of in jacob's layout PR #784  👍

File: src/components/BookmarkList/BookmarkList.tsx (hunk 2 above)
Comment (abbyoung) [reply src/components/BookmarkList/BookmarkList.tsx:116]: `<Table>` is a react uswds component and wouldn't accept a `className` prop. If I missed something and there's a way to pass one, please let me know!

Comment (gidjin) [reply src/components/BookmarkList/BookmarkList.tsx:116]: Sadly I don't know all the React-uswds components well enough yet. But I checked and looks like it does not accept any classes. I'm not sure if it should or not, but does answer my question :)

**Review by gidjin — APPROVED** [review]
LGTM!

**Pull request merged**